src/components/tips/tip-uploader.tsx

```diff
@@ -0,0 +1,301 @@
+    createTip({
+      ...values,
+      fileName: slugify(values.title.toLowerCase(), {remove: /[*+~.()'"!:@]/g}),
+    })
+  }
+
+  return (
+    <Form {...form}>
+      <form onSubmit={form.handleSubmit(onSubmit)} className="space-y-8">
+        <FormField
+          control={form.control}
+          name="title"
+          render={({field}) => (
+            <FormItem>
+              <FormLabel className="text-lg font-bold">Title*</FormLabel>
+              <FormDescription className="mt-2 text-sm">
+                Include any relavent keywords for SEO.
+              </FormDescription>
+              <FormControl>
+                <Textarea placeholder="How do I..." {...field} />
+              </FormControl>
+
+              <FormMessage />
+            </FormItem>
+          )}
+        />
+        <FormField
+          control={form.control}
+          name="description"
+          render={({field}) => (
+            <FormItem>
+              <FormLabel className="text-lg font-bold">Description</FormLabel>
+              <FormDescription className="mt-2 text-sm">
+                This should describe what you teach in the tip and include any
+                relevant codeblocks and links.
+              </FormDescription>
+              <FormControl>
+                <Textarea
+                  placeholder="add as much detail as you'd like :)"
+                  {...field}
+                />
+              </FormControl>
+
+              <FormMessage />
+            </FormItem>
+          )}
+        />
+
+        {s3Url.length === 0 ? (
+          <TipUploader
+            sets3Url={(s3Url: string) => {
+              form.setValue('s3Url', s3Url)
+              sets3Url(s3Url)
+            }}
+          />
+        ) : (
+          <div>
+            {s3Url}
+            <Button
+              onClick={() => {
+                sets3Url('')
+                form.setValue('s3Url', '')
+              }}
+            >
+              clear
+            </Button>
+          </div>
+        )}
+        <FormField
+          control={form.control}
+          name="s3Url"
+          render={({field}) => (
+            <FormItem>
+              <FormControl>
+                <Input type="hidden" {...field} />
+              </FormControl>
+              {form.formState.errors.s3Url && (
+                <p>{form.formState.errors.s3Url.message}</p>
+              )}
+            </FormItem>
+          )}
+        />
+        <Button
+          type="submit"
+          className="mt-2"
+          variant="default"
+          onClick={() => {
+            if (!s3Url) {
+              form.setError('s3Url', {
+                type: 'manual',
+                message: 'Please upload a video',
+              })
+            }
+          }}
+        >
+          Submit Tip
+        </Button>
+      </form>
+    </Form>
+  )
+}
+
+function TipUploader({sets3Url}: {sets3Url: (value: string) => void}) {
```

Comment:
there was a lot of logic here which is the original S3 uploader so put it in it's own component